### PR TITLE
TRON for quasi-Newton models

### PR DESCRIPTION
**Pull request by dpo**
Same deal. These are the results on the same unconstrained problems. Going to test bound-constrained problems.

tronqn

### CODE REVIEW

**Comment by dpo** [comment]
59 bound-constrained problems from CUTEst:

tronqn_bnds

File: src/tron.jl [hunk 1]
```diff
@@ -148,7 +158,7 @@ function tron(::Val{:Newton},
   end
 
   return GenericExecutionStats(status, nlp, solution=x, objective=fx, dual_feas=πx,
-                               iter=iter, elapsed_time=el_time)
+                               primal_feas=0, iter=iter, elapsed_time=el_time)
```
Comment (abelsiqueira) [inline src/tron.jl]: `primal_feas` wasn't automatically 0? Also, it needs to be a `zero(eltype(x))`.

Comment (dpo) [reply src/tron.jl]: It was at Inf!

Comment (abelsiqueira) [reply src/tron.jl]: Ah, it's `0` if `unconstrained(nlp)`.

**Pull request merged**